### PR TITLE
[REVIEW] Drop python `3.7` in code-base

### PR DESCRIPTION
**Pull request by galipremsagar**
Starting `22.02` we dropped support for python `3.7`: https://docs.rapids.ai/notices/rsn0014/, but there are places in code-base that still have 3.7 usages. This PR removes and upgrades those usages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,30 +65,30 @@ Please see the [Demo Docker Repository](https://hub.docker.com/r/rapidsai/rapids
 
 cuDF can be installed with conda ([miniconda](https://conda.io/miniconda.html), or the full [Anaconda distribution](https://www.anaconda.com/download)) from the `rapidsai` channel:
 
-For `cudf version == 21.08` :
+For `cudf version == 22.06` :
 ```bash
 # for CUDA 11.0
 conda install -c rapidsai -c nvidia -c numba -c conda-forge \
-    cudf=21.08 python=3.7 cudatoolkit=11.0
+    cudf=21.08 python=3.8 cudatoolkit=11.0
```
Comment (bdice) [inline README.md:72]: Since we officially support 3.9, can we future-proof this a bit more and go straight to 3.9?

**Review by bdice — APPROVED** [review]
LGTM but we should use `python=3.9` in the example commands.

File: README.md (hunk 1 above)
Comment (galipremsagar) [inline README.md:72]:
```suggestion
    cudf=21.08 python=3.9 cudatoolkit=11.0
```

**Comment by bdice** [comment]
Another related change would be to update all the `pyproject.toml` files so that the `black` formatter is targeting `py38`. There are 5 such cases: https://github.com/rapidsai/cudf/search?q=py36

**Comment by bdice** [comment]
@galipremsagar We need the black formatter to target `py38`. If it targets `py39`, it could apply formatting changes that are supported only by Python 3.9+, while we need Python 3.8 compatibility. I don't know if this makes any difference in practice, but the _minimum_ supported version is what I would recommend.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**